Feat(client/ C#): Update TopologyRequestCommand and UpdateRetriesCommand for retries

askalik opened the pull request:
closes #184 , #183 


Updated the interfaces, methods, and main application to use asyncRetryStrategy. 

Please let me know if I have missed anything

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda-community-hub/zeebe-client-csharp?pullRequest=329) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/camunda-community-hub/zeebe-client-csharp?pullRequest=329) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda-community-hub/zeebe-client-csharp?pullRequest=329) it.</sub>

Zelldon commented on Client/Impl/Commands/UpdateRetriesCommand.cs:
```diff
@@ -35,9 +36,9 @@ public async Task<IUpdateRetriesResponse> Send(TimeSpan? timeout = null, Cancell
         }
 
-        public async Task<IUpdateRetriesResponse> Send(CancellationToken cancellationToken)
+        public async Task<IUpdateRetriesResponse> SendWithRetry(CancellationToken cancellationToken)
```
It seems you removed the send method, could you please bring it back, such that we have two sends and one sendWithRetry. Like here https://github.com/camunda-community-hub/zeebe-client-csharp/blob/master/Client/Impl/Commands/CompleteJobCommand.cs#L49-L64

askalik replied:
brought back the send so that there are two sends and a send with retry

Zelldon reviewed (commented):
Hey @askalik thanks so much for your contribution :) 

Could you please create two separate PR's next time, to separate the concerns :) 

You currently have two commits, please squash them (the first one has no real change).

Sorry, I had to forget to mention, we also need to test this. For that please add also a TestCaseData object on this provider https://github.com/camunda-community-hub/zeebe-client-csharp/blob/master/Client.UnitTests/TestDataProvider.cs, such that the parameterized test is running also for this command. Feel free to ask any questions you might have.

askalik commented:
@Zelldon  I squashed the commits,  and re-added the send method. 

Working on getting the test updated.  Is it the topology that is to be added, updated with retries - or both?

Zelldon commented:
Thanks @askalik 

yes please add two new test case data objects here https://github.com/camunda-community-hub/zeebe-client-csharp/blob/master/Client.UnitTests/TestDataProvider.cs#L13


Example for the topology:
```
yield return new TestCaseData(
                    new TopologyRequest
                    {
                        ProcessInstanceKey = 12113
                    }, new TopologyResponse(),
                    (RequestCreator<ITopology>)
                    (zeebeClient => zeebeClient.TopologyRequest()));

```

Zelldon commented on Client.UnitTests/TestDataProvider.cs:
```diff
@@ -33,6 +33,21 @@ public static IEnumerable<TestCaseData> Provider()
+                yield return new TestCaseData(
+                    new TopologyRequest
+                    {
+                        ProcessInstanceKey = 12113
```
Sorry my example was bit wrong :see_no_evil: here the key makes no sense, I'm wondering that this compiles :laughing: 

But it is fine I can fix it later :) I will merge it anyway Thanks @askalik for all your efforts :rocket:

Zelldon reviewed (approved):
:medal_military:
